File: app/src/types/request/player.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import type { WorldId } from '@/types/core/world';
import type { PlayerId, PlayerOptions, PlayerStatus } from '@/types/core/player';

export interface GetPlayerRequest {
  readonly world: WorldId;
}

export interface GetPlayerCoordsRequest {
  readonly world: WorldId;
  readonly id: PlayerId;
}

export interface GetPlayerIdsRequest {
  readonly world: WorldId;
}

export interface GetPlayerMaintenanceRequest {
  readonly world: WorldId;
}

export interface GetPlayerMilitaryRequest {
  readonly world: WorldId;
}

export interface GetPlayerReportsRequest {
  readonly world: WorldId;
}

export interface GetPlayerStatusRequest {
  readonly world: WorldId;
  readonly id: PlayerId;
}

export interface GetPlayerStorageCapacityRequest {
  readonly world: WorldId;
}

export interface GetPlayerWorldsRequest {
  readonly id: PlayerId;
}

export interface GetPublicPlayerRequest {
  readonly world: WorldId;
  readonly id: PlayerId;
}

export interface GetPublicPlayersRequest {
  readonly world: WorldId;
}

export interface PlayerExistsRequest {
  readonly world: WorldId;
  readonly id: PlayerId;
}

export interface SetPlayerStatusRequest {
  readonly world: WorldId;
  readonly status: PlayerStatus;
}

export interface SpawnPlayerRequest {
  readonly world: WorldId;
  readonly worldPassword: Option<string>;
  readonly options: PlayerOptions;
}

File: crates/nil-payload/src/request/player.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::player::{PlayerId, PlayerOptions, PlayerStatus};
use nil_core::world::config::WorldId;
use nil_crypto::password::Password;
use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPlayerRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPlayerCoordsRequest {
  pub world: WorldId,
  pub id: PlayerId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPlayerIdsRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPlayerMaintenanceRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPlayerMilitaryRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPlayerReportsRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPlayerStatusRequest {
  pub world: WorldId,
  pub id: PlayerId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPlayerStorageCapacityRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPlayerWorldsRequest {
  pub id: PlayerId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPublicPlayerRequest {
  pub world: WorldId,
  pub id: PlayerId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct GetPublicPlayersRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct PlayerExistsRequest {
  pub world: WorldId,
  pub id: PlayerId,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct SetPlayerStatusRequest {
  pub world: WorldId,
  pub status: PlayerStatus,
}

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export, optional_fields = nullable)]
pub struct SpawnPlayerRequest {
  pub world: WorldId,
  #[serde(default)]
  pub world_password: Option<Password>,
  pub options: PlayerOptions,
}
